src/main/scala/com/microsoft/ml/spark/vw/VowpalWabbitBase.scala

```diff
@@ -166,13 +166,23 @@ trait VowpalWabbitBase extends Wrappable
   protected def createLabelSetter(schema: StructType) = {
     val labelColIdx = schema.fieldIndex(getLabelCol)
 
+    // support numeric types as input 
+    def getAsFloat(idx: Int) =
+      schema.fields(idx).dataType match {
+        case _: DoubleType => (row: Row) => row.getDouble(idx).toFloat
```

Comment:
should we warn about loss of fidelity?

Reply:
I wouldn't as you can't do anything against it as VW native doesn't support double types

Reply:
makes sense, hence why warn instead of throw imo - silent loss seems sad

just an idle midnight thought :D